utils/utils.go

```diff
@@ -1,4 +1,4 @@
-package lifecycle
+package utils
```

Comment:
I like your idea. I don't like putting everything together in `utils` as well.
I'll let you know after I'll make some changes.
Thanks!

Reply:
Please let me know what you think about the new packages in my last commit: https://github.com/buildpacks/lifecycle/pull/724/commits/3917722d377eb5c719ff6bd97ef5c155e8a7191c.
@jabrown85 @natalieparellano

Reply:
I have to say that I'm still a bit wary of creating all these tiny packages. I did a little digging to try to substantiate this feeling, and found this [post](https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common):

> My recommendation to improve the name of utils or helpers packages is to analyse where they are imported and move the relevant functions into the calling package. Even if this results in some code duplication this is preferable to introducing an import dependency between two packages. In the case where utility functions are used in many places, prefer multiple packages, each focused on a single aspect with a correspondingly descriptive name.

We are doing the "multiple focused packages" part, but all of the utility functions that have been copied are used only in the `lifecycle` package - with the exception of `WriteTOML`. I think I'd still rather duplicate this function (it's very simple, and only 11 lines of code) than introduce all of these new packages with no clear need to do so.

Reply:
I don't like code duplication unless the function is almost an inline function, and even then, you really need to think whether it's worth it.

Talking about `WriteTOML`, if at some point we'll find a bug in this function or we'll decide to change it a bit, having to do it in 3 different (or even 2) places will increase the probability of new bugs because we might forget to change it everywhere or we might copy it incorrectly.

Regarding the the package`imageutils` - it is being used by 3 other packages: `lifecycle`, `cache` and `main` so I would leave at least the `DecodeLabel` function there.

The package `str` is only being used by the `lifecycle` package so I'm ok with moving its functions to somewhere there.

Reply:
I removed a line at the end of my comment that was originally like "if we don't like small packages I rather have some duplication". I think there is a balance to be struck here for sure. I think I lean toward small packages but I don't have very strong feelings.

Reply:
> Regarding the the packageimageutils - it is being used by 3 other packages: lifecycle, cache and main so I would leave at least the DecodeLabel function there.

Fair enough. But let's not make `imageutils` (too similar to `imgutil` lol) - I noticed we already have an `image` package, so maybe it could go there.

> The package str is only being used by the lifecycle package so I'm ok with moving its functions to somewhere there.

Let's just keep it in utils.go then.

> I don't like code duplication unless the function is almost an inline function, and even then, you really need to think whether it's worth it.

I think we'll just have to agree to disagree :) Can we please keep both `lifecycle.WriteTOML` and `buildpack.WriteTOML` for now - we know that we have a lot of work to do re: reading and writing TOML and perhaps new packages will emerge from that effort. If you want you could write a test that `lifecycle.WriteTOML` and `buildpack.WriteTOML` produce the same output.

Reply:
I moved the imageutils files to the image package as part of the following commit:
https://github.com/buildpacks/lifecycle/pull/724/commits/6559c5c07ee41e11c6b8f553949bd43b9b93b325

Regarding `WriteTOML`:
You want to define the function in both the `lifecycle` and the `buildpack` packages, right?
But what about `layer_metadata_restorer.go`? It can't use the function that will be define in the `lifecycle` package because it will cause circular imports. Moreover, as you wrote in the past about the `lifecycle` package, it doesn't make sense that the `layermetadata` package will use the `buildpack` package for that function.
So what do you want to do? Define it again for the third time?
@natalieparellano @jabrown85

Reply:
I think in the end, we’ll probably have two packages with toml utilities - one for platform, and one for buildpack (since the way we are parsing & interpreting files will differ by each api). Maybe it makes sense to explore what that will look like as part of this PR?

Reply:
I'm not sure I understand what is the problem that there will be a common package with useful functions.
I don't think that `WriteTOML` will depend on an API version.

Reply:
Following the discussion in the [slack](https://buildpacks.slack.com/archives/CSFE49VGR/p1633451965101500), please see https://github.com/buildpacks/lifecycle/pull/724/commits/3caa11617113824dcdd2aa082db932963c887932.